### ISSUE TITLE
[HLS] No audio with .aac

### ISSUE THREAD

**Issue by matthuisman** [peak3d/inputstream.adaptive#686]
Below stream will play with video and subtitles but no audio.
Kodi shows 0 channels aac.

Master manifest: http://cdn.theoplayer.com/video/elephants-dream/playlist-single-audio.m3u8
Audio manifest: http://cdn.theoplayer.com/video/elephants-dream/en/chunklist_w370587926_b160000_ao_slen_t64RW5nbGlzaA==.m3u8
Video manifest: http://cdn.theoplayer.com/video/elephants-dream/1280/chunklist_w370587926_b2962000_vo_slen_t64TWFpbg==.m3u8
Test strm file: https://raw.githubusercontent.com/matthuisman/ia_tests/master/hls_subtitles/hls_subtitles.strm

[kodi.log.txt](https://github.com/xbmc/inputstream.adaptive/files/6488894/kodi.log.txt)

it only ever downloads the first aac segment.

ignore the /VIDEO_TS.IFO stuff.
Thats a bug in 19.1 that I have a PR fix for here: https://github.com/xbmc/xbmc/pull/19776

**Comment by glennguy** [comment]
Hey I've had a quick look at this last night. The pts/offset values for the audio stream are being set to some astronomical value on stream start. My guess what happens then is the reader wants to find the right (0) pts value but as far as it can see it doesn't exist, so no way to seek to it. I opened in the debugger and set pts for the ADTS reader to 0 and stream then plays with audio.

Need to investigate further

**Comment by glennguy** [comment]
Do you have any other streams to test with aac audio? I think I have a fix.

**Comment by matthuisman** [comment]
Yes, found one!
https://storage.googleapis.com/shaka-demo-assets/apple-advanced-stream-ts/master.m3u8

```
#KODIPROP:inputstream=inputstream.adaptive
#KODIPROP:inputstreamaddon=inputstream.adaptive
#KODIPROP:inputstream.adaptive.manifest_type=hls
https://storage.googleapis.com/shaka-demo-assets/apple-advanced-stream-ts/master.m3u8
```

confirmed no audio on above either

**Comment by glennguy** [comment]
This one doesn't play for a different reason :) there's always something!

**Comment by matthuisman** [comment]
I'm sorry 😔

**Comment by glennguy** [comment]
I've been testing again on the elephant stream and found my fix was not good (seeking around causes lots of problems). Also, if you resume the stream the audio works fine with code as-is. I wish I knew more about ADTS...

**Comment by CastagnaIT** [comment]
fixed by #1241

instead for the other problematic sample
https://storage.googleapis.com/shaka-demo-assets/apple-advanced-stream-ts/master.m3u8
i copied it to the appropriate issue #1085
